src/kaboom.ts

```diff
@@ -279,6 +281,37 @@ function createEmptyAudioBuffer(ctx: AudioContext) {
     return ctx.createBuffer(1, 1, 44100);
 }
 
+type InternalContext = {
+    kaboomCtx: KaboomCtx;
+    getViewportScale: () => number;
+};
+
+let ctx: KaboomCtx;
+let internalCtx: InternalContext[] = [];
+
+// TODO: A better way to detect KaboomCtx
+export const isKaboomCtx = (obj: any): obj is KaboomCtx => {
+    return Object.keys(obj).includes("loadAseprite");
```

Comment:
This line doesn't look very nice. It can be tiring to convert the object to an array and check the keys one by one each time. If a component is added and removed frequently, this process will be repeated each time.

Reply:
Yeah, maybe only check obj["loadAseprite"]